packages/common/src/types/savedCharts.ts

```diff
@@ -305,3 +315,37 @@ export const isSeriesWithMixedChartTypes = (
                 }`,
         ),
     ).size >= 2;
+
+export const getChartType = (
+    chartType: ChartType,
+    value: ChartConfig['config'],
+): ChartKind | undefined => {
+    if (value === undefined) return undefined;
+    switch (chartType) {
+        case ChartType.BIG_NUMBER:
+            return ChartKind.BIG_NUMBER;
+        case ChartType.TABLE:
+            return ChartKind.TABLE;
+        case ChartType.CARTESIAN:
+            if (isCartesianChartConfig(value)) {
+                if (isSeriesWithMixedChartTypes(value.eChartsConfig.series))
+                    return ChartKind.MIXED;
+
+                switch (value.eChartsConfig.series?.[0]?.type) {
+                    case CartesianSeriesType.AREA:
+                        return ChartKind.AREA;
+                    case CartesianSeriesType.BAR:
+                        return ChartKind.BAR;
+                    case CartesianSeriesType.LINE:
+                        return ChartKind.LINE;
+                    case CartesianSeriesType.SCATTER:
+                        return ChartKind.SCATTER;
+                    default:
+                        return undefined;
+                }
+            }
+            return undefined;
+        default:
+            return undefined;
```

Comment:
I think it would be good if you say before `switch` that if `value.eChartsConfig.series?.[0]?.type` does not exist then return `undefined` and then default to `assertUnreachable` to "exhaust" the switch case.

ideally we should have switch cases that always "exhaust" all possible options